### PR TITLE
Implement game

### PR DESCRIPTION
**Pull request by omarhurani**
## What?
Added the implementation to the ping pong game including:
- Round ball 🔴
- Two controllable paddles, one with the `W` and `S` keys and the other with the `↑` and `↓`  keys, for each player.
- Pause/resume using the **Space** key ⏸
- Score board in the middle of the screen for each player 💯
- Adaptive size, where the game keeps a fixed aspect ratio and always fits the browser window.
- Sound effects 🔊

## Why?
In order to join the awesome team of Colonist.io 😁

## How?
- No external libraries, pure HTML5, CSS and Javascript.
- Minimal modifications to `app.js`.
- Divided the code into multiple classes and used polymorphism to define default behaviors and override them when necessary.
- Lots of help from [MDN](https://developer.mozilla.org/en-US/) and [Stackoverflow](https://stackoverflow.com/).

## Video
https://user-images.githubusercontent.com/26940488/154147153-5ad290fd-379d-41b6-81e1-ed9d5a75879c.mp4

### CODE REVIEW

File: public/js/Ball.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        super(id, {x, y}, 10, 'red', speed, direction);
+        this.bounceAudio = bounceAudio;
+    }
+
+    _bounce({minY = 0, maxY = Number.MAX_VALUE}){
+        // If this.y is at the minY or maxY, reverse the Y direction
+        if(this.y <= minY || (this.y + this.height) >= maxY ){
+            this.direction = -this.direction;
+            this.bounceAudio?.play();
+        }
```
Comment (demiculus) [inline public/js/Ball.js]: This code should be refactored as such
```
const aDescriptiveVariableName = this.y <= minY || (this.y + this.height) >= maxY
if(aDescriptiveVariableName) {
  this.direction = -this.direction
  this.bounceAudio?.play();
}
```

A better way is 

```
const aDescriptiveVariableName = this.y <= minY || (this.y + this.height) >= maxY
if(aDescriptiveVariableName) aDescriptiveFunctionName()

function aDescriptiveFunctionName() {
  this.direction = -this.direction
  this.bounceAudio?.play();
}
```

**Comment by demiculus** [comment]
@omarhurani well done the game looks great. 
The code can be improved. Code readability is one of the most important thing for non-throwaway-project.

The best code is a code that can be understood by English without going into math. In other words an English reader should be able to read the code just like reading a text book. I gave an example of this here: https://github.com/omarhurani/colonist_test/pull/1/files#r807518130

So knowing these can you create a new branch on top of this and do the necessary changes to the code?

**Comment by omarhurani** [comment]
Thanks for the feedback!
Absolutely, I'll work on it.

**Comment by Esqarrouth** [comment]
well done, watching the game put a smile on my face

**Pull request merged**